_posts/2019-01-22-putting-phoenix-in-an-iframe.md

```diff
@@ -10,6 +10,50 @@ We recently had cause to run a Phoenix application within an iframe on another a
 
 In order to get our application to display we needed to remove the 'X-Frame-Options' header from the response we were sending. In Phoenix this header (among others) is added by default through a plug in the router, [put_secure_browser_headers/2](https://hexdocs.pm/phoenix/Phoenix.Controller.html#put_secure_browser_headers/2).
 
-We need to be careful though. This header is there for [our protection](https://responsivedesign.is/articles/xframe-options/) to prevent [clickjacking](https://www.owasp.org/index.php/Clickjacking), so we should only remove the header after authenticating the source of the request. We decided to create a custom plug that would authenticate the request and then remove the header from the response on success. Phoenix has a handy function [delete_resp_header/2](https://hexdocs.pm/plug/Plug.Conn.html#delete_resp_header/2) that lets us do just that.
+We need to be careful though. This header is there for [our protection](https://responsivedesign.is/articles/xframe-options/) to prevent [clickjacking](https://www.owasp.org/index.php/Clickjacking), so we should only remove the header after authenticating the source of the request.
+
+To solve this we created a custom plug that would authenticate the request and then remove the header from the response on success. Phoenix has a handy function [delete_resp_header/2](https://hexdocs.pm/plug/Plug.Conn.html#delete_resp_header/2) that lets us do just that.
+
+```elixir
+# controllers/allow_cross_origin_iframe.ex
+defmodule CultivateWeb.Example do
+  import Plug.Conn
+  def init(opts), do: opts
+
+  def call(conn, _options) do
+    conn
+    |> authenticate_source
```

Comment:
Nice to have source.

Parenthesis needed here, though. `authenticate_source()`

Reply:
(Actually never mind - I have it open)

Reply:
Thanks!